docs/xks/developer-guide/container-security.md

```diff
@@ -0,0 +1,293 @@
+            - NEW_RAW
+```
+
+### Proc Mount
+
+There are certain files in the `/proc` directory that may be sensitive is exposed to a container. By default the container runtime will apply a filter the files mounted to the container from the directory, so opting out of it is not allowed.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: app
+spec:
+  containers:
+    - securityContext:
+        procMount: UnmaskedProcMount 
+```
+
+### Volume Type
+
+The types of volumes are limited to a known set to mitigate host path mounting and the use of unknown volume types. The allowed volume types cover the majority of use cases. The following volume types are currently allowed in XKF, `configMap`, `downwardAPI`, `emptyDir`, `persistentVolumeClaim`, `secret`, `projected`, `csi`.
+
+As stated before blocking any `hostPath` volumes is critical as mounting certain directories or files on the host would give a container insight to or control over the host.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: app
+spec:
+  volumes:
+    - hostPath:
+        path: /proc
+```
+
+### Flex Volumes
+
+[Flex Volumes](https://kubernetes.io/docs/concepts/storage/volumes/#flexvolume) are a deprecated type of volume that are no longer maintained in the Kubernetes project. Flex volume drives are not installed in XKF clusters but are still not allowed to be used.
+
+## Non Root User
+
+A common practice for a lot of public images is to, if not otherwise specified, run the containers as root with UID 0. The main reason this is done is because it is the default in Docker. Changing to another user by default has to be opted in. Running the container as root is generally not a problem as it is not the same as root on the node. It does however become a problem when new container vulnerabilities are found which allow escaping the sandbox. An example of such a vulnerability is [CVE-2019-5736](https://avd.aquasec.com/nvd/2019/cve-2019-5736/) which allowed containers running as root users to escalate its privileges to becoming root on the node. One of the mitigations for this vulnerability is to make sure that all containers run as non root.
+
+With the knowledge that the mitigation for future container vulnerabilities is so simple it would be lazy not to take the precautionary steps. One method to change the user used to run the container is to do it at the image source. It is as simple as changing the user. Some testing may have to be done initially if the applications needs to read files which have different permissions. The example below sets both the user and group to `65534` which is an existing user called `nouser`.
+
+```Dockerfile
+USER 65534:65534
+```
+
+Another solution is to make changes to the Pod so that a different user is used when running. The user id and group id used can be set in the Pods security context.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: app
+spec:
+  securityContext:
+    runAsUser: 65534
+    runAsGroup: 65534
+```
```

Comment:
Not sure about all these recommendations. Would love others input so that we get this right for the future when this becomes enforced.

Reply:
I think it's good ideas. The problem is how should you handle things like node that run on USER 1000 by default on most images.
And my guess is that node would actually get issues if you tried to run this config with `runAsUser`.
I think this is a okay start but I think we need to give some example and maybe say that setting user 1000 is better then not setting it all.